protocols/kad/src/behaviour.rs

```diff
@@ -399,6 +402,14 @@ impl KademliaConfig {
         self.caching = c;
         self
     }
+
+    /// Sets the mode.
+    ///
+    /// The default is [`Mode::Server`].
+    pub fn set_mode(&mut self, m: Mode) -> &mut Self {
+        self.mode = m;
+        self
+    }
```

Comment:
The `Mode` needs to be changeable at runtime. I expect the most common usage to be a call to `set_mode` based on an at-runtime `libp2p-autonat` result.